identity-account/src/updates/update.rs

```diff
@@ -0,0 +1,465 @@
+use identity_did::verification::MethodRelationship;
+use identity_did::verification::MethodScope;
+use identity_did::verification::MethodType;
+use identity_iota::did::IotaDID;
+use identity_iota::did::IotaDIDUrl;
+use identity_iota::did::IotaDocument;
+use identity_iota::did::IotaVerificationMethod;
+
+use crate::account::Account;
+use crate::error::Result;
+use crate::identity::DIDLease;
+use crate::identity::IdentitySetup;
+use crate::identity::IdentityState;
+use crate::storage::Storage;
+use crate::types::Generation;
+use crate::types::KeyLocation;
+use crate::types::MethodSecret;
+use crate::updates::UpdateError;
+
+// Method types allowed to sign a DID document update.
+pub const UPDATE_METHOD_TYPES: &[MethodType] = &[MethodType::Ed25519VerificationKey2018];
+pub const DEFAULT_UPDATE_METHOD_PREFIX: &str = "sign-";
+
+fn key_to_method(type_: KeyType) -> MethodType {
+  match type_ {
+    KeyType::Ed25519 => MethodType::Ed25519VerificationKey2018,
+  }
+}
+
+pub(crate) async fn create_identity(setup: IdentitySetup, store: &dyn Storage) -> Result<(DIDLease, IdentityState)> {
+  let method_type = key_to_method(setup.key_type);
+
+  // The method type must be able to sign document updates.
+  ensure!(
+    UPDATE_METHOD_TYPES.contains(&method_type),
+    UpdateError::InvalidMethodType(method_type)
+  );
+
+  let generation = Generation::new();
+  let fragment: String = format!("{}{}", DEFAULT_UPDATE_METHOD_PREFIX, generation.to_u32());
+
+  let location: KeyLocation = KeyLocation::new(method_type, fragment, generation);
+
+  let keypair: KeyPair = if let Some(MethodSecret::Ed25519(private_key)) = &setup.method_secret {
+    ensure!(
+      private_key.as_ref().len() == ed25519::SECRET_KEY_LENGTH,
+      UpdateError::InvalidMethodSecret(format!(
+        "an ed25519 private key requires {} bytes, found {}",
+        ed25519::SECRET_KEY_LENGTH,
+        private_key.as_ref().len()
+      ))
+    );
+
+    KeyPair::try_from_ed25519_bytes(private_key.as_ref())?
+  } else {
+    KeyPair::new_ed25519()?
+  };
+
+  // Generate a new DID from the public key
+  let did: IotaDID = if let Some(network) = &setup.network {
+    IotaDID::new_with_network(keypair.public().as_ref(), network.clone())?
+  } else {
+    IotaDID::new(keypair.public().as_ref())?
+  };
+
+  ensure!(
+    !store.key_exists(&did, &location).await?,
+    UpdateError::DocumentAlreadyExists
+  );
+
+  let did_lease = store.lease_did(&did).await?;
+
+  let private_key = keypair.private().to_owned();
+  std::mem::drop(keypair);
+
+  let public: PublicKey =
+    insert_method_secret(store, &did, &location, method_type, MethodSecret::Ed25519(private_key)).await?;
+
+  let method_fragment = location.fragment().to_owned();
+
+  // TODO: Can we unwrap/expect?
+  let method: IotaVerificationMethod =
+    IotaVerificationMethod::from_did(did, setup.key_type, &public, method_fragment.name())?;
+
+  // TODO: Can we unwrap/expect?
+  let document = IotaDocument::from_verification_method(method)?;
```

Comment:
There may be unforeseen errors in the future, safer to use `?` or map it to some other error if desired.